nifi-nar-bundles/nifi-enrich-bundle/nifi-enrich-processors/src/main/java/org/apache/nifi/processors/enrich/QueryDNS.java

```diff
@@ -0,0 +1,272 @@
+        props.add(DNS_RETRIES);
+        props.add(DNS_TIMEOUT);
+        props.add(DNS_SERVER);
+        props.add(DNS_QUERY_TYPE);
+        propertyDescriptors = Collections.unmodifiableList(props);
+
+        Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_FOUND);
+        rels.add(REL_NOT_FOUND);
+        relationships = Collections.unmodifiableSet(rels);
+    }
+
+    private AtomicBoolean initialized = new AtomicBoolean(false);
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        if (!initialized.get()) {
+            initializeResolver(context);
+            getLogger().warn("Resolver was initialized at onTrigger instead of onScheduled");
+
+        }
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String queryType = context.getProperty(DNS_QUERY_TYPE).getValue();
+        final String queryInput = context.getProperty(QUERY_INPUT).evaluateAttributeExpressions(flowFile).getValue();
+        final String queryParser = context.getProperty(QUERY_PARSER).getValue();
+        final String queryRegex = context.getProperty(QUERY_PARSER_INPUT).getValue();
+
+        boolean found = false;
+        try {
+            Attributes results = doLookup(queryInput, queryType);
+            // NOERROR & NODATA seem to return empty Attributes handled bellow
+            // but defaulting to not found in any case
+            if (results.size() < 1) {
+                found = false;
+
+
+            } else {
+                int recordNumber = 0;
+                NamingEnumeration<?> dnsEntryIterator = results.get(queryType).getAll();
+                while (dnsEntryIterator.hasMoreElements()) {
+                    String dnsRecord = dnsEntryIterator.next().toString();
+                    // While NXDOMAIN is being generated by doLookup catch
+                    if (dnsRecord != "NXDOMAIN") {
+                        Map<String, String> parsedResults = parseResponse(recordNumber, dnsRecord, queryParser, queryRegex, "dns");
+                        flowFile = session.putAllAttributes(flowFile, parsedResults);
+                        found = true;
+
+
+                    } else {
+                        // Otherwise treat as not found
+                        found = false;
+
+
+                    }
+                    // Increase the counter and iterate over next record....
+                    recordNumber++;
+                }
+            }
+        } catch (NamingException e) {
+            session.rollback(true);
+            getLogger().error("Something went wrong while processing the field. Please review your configuration");
```

Comment:
addressed. I removed the voluntary rollback. Instead we now throw a ProcessorException